### ISSUE TITLE
Subsequent back-to-back deployments result in a failure 

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [JamesIves/github-pages-deploy-action#679]
I am publishing docs to multiple folders.

- Latest snapshot version to `snapshots`
- Specific versions to `versions/some-version`, e.g. `versions/0.1.0-alpha.14`

For some reason, snapshot version is working perfectly, but the specific version no.

The same setup, even the reports are the same.

As you can see, created folders are correct too:

```
/usr/bin/rsync -q -av --checksum --progress /home/runner/work/gradle-plugins/gradle-plugins/build/dokka/. github-pages-deploy-action-temp-deployment-folder/snapshot --exclude .ssh --exclude .git --exclude .github
```

```
/usr/bin/rsync -q -av --checksum --progress /home/runner/work/gradle-plugins/gradle-plugins/build/dokka/. github-pages-deploy-action-temp-deployment-folder/versions/0.1.0-alpha.15 --exclude .ssh --exclude .git --exclude .github
```

Workflows:
- [Snapshots](https://github.com/JavierSegoviaCordoba/gradle-plugins/actions/runs/775248082/workflow)
```
      - name: Deploy snapshot docs to Github Pages
        uses: JamesIves/github-pages-deploy-action@4.1.0
        with:
          target-folder: 'snapshot'
          branch: 'gh-pages'
          folder: 'build/dokka'
          clean: false
          single-commit: true
```
- [Specific version](https://github.com/JavierSegoviaCordoba/gradle-plugins/actions/runs/775265361/workflow)
```
      - name: Deploy release docs to Github Pages inside specific version folder
        uses: JamesIves/github-pages-deploy-action@4.1.0
        with:
          target-folder: 'versions/${{ steps.tags.outputs.version }}'
          branch: 'gh-pages'
          folder: 'build/dokka'
          clean: false
          single-commit: true
```

What is happening? I think it is impossible `build/dokka` is empty.

Maybe single-commit?

**Comment by JamesIves** [comment]
The message you're getting does stem from the single-commit option, but even without using it you're going to see another message related to the orphan branch `gh-pages` already existing. I think there's a degree of orphan branch pruning that needs to occur. I'm going to leave this ticket open while I investigate this as I agree this is pretty annoying.

In the interim you can simply run the checkout step and your build script again before making the second deployment.

**Comment by JavierSegoviaCordoba** [comment]
But, indeed, only one deployment is happening, because I am releasing alpha versions, so the second one is omitted in the same job.

**Comment by JamesIves** [comment]
I've got a working prototype of this running on the `releases/v4-alpha` branch if you'd like to test it out. All you need to do is point your workflow to `JamesIves/github-pages-deploy-action@releases/v4-alpha`.

I still have a number of things to wrap up before 4.1.2 will be released but wanted to give you a preview to test out in the interim. Let me know if it works as expected!

**Comment by JavierSegoviaCordoba** [comment]
I will try to test it so I can give you feedback, hope before this week ends.

**Comment by JavierSegoviaCordoba** [comment]
Tested and working, great job 👏 (logs [here](https://github.com/JavierSegoviaCordoba/massive-catalogs/runs/2459612666?check_suite_focus=true))

I reverted the changes in the repo to wait for 4.1.2.

**Comment by JavierSegoviaCordoba** [comment]
Can you take a look to [this one](https://github.com/JavierSegoviaCordoba/gradle-plugins/runs/2468904591?check_suite_focus=true)? 

As you can see, `_site` folder has a lot files, but I am not able to commit them

**Comment by JavierSegoviaCordoba** [comment]
Anyway, I removed GitHub pages branch, and rebuilding it works

**Comment by JamesIves** [comment]
Released as part of 4.1.2.
